scripts/verifyChange.ts

```diff
@@ -0,0 +1,120 @@
+import util from 'node:util';
+
+type Options = {
+  fileLineChange: string;
+  diffPatchSeparated: string;
+  verbose: boolean;
+};
+
+function parseOptions(): Options {
+  const options = {
+    fileLineChange: { type: 'string' },
+    diffPatchSeparated: { type: 'string' },
+    verbose: { type: 'boolean' },
+  } as const;
+  const { values } = util.parseArgs({ options });
+
+  for (const option of ['fileLineChange', 'diffPatchSeparated'] as const) {
+    if (values[option] === undefined) {
+      throw new Error(`missing --${option} flag`);
+    }
+  }
+
+  return {
+    ...(values as Options),
+    verbose: values.verbose === true,
+  };
+}
+
+async function main() {
+  const options = parseOptions();
+  const debug = (...msg: unknown[]) => {
+    if (options.verbose) {
+      console.debug(...msg);
+    }
+  };
+
+  const fileLineChangeJSON = JSON.parse(options.fileLineChange);
+  const diffPatchSeparated = Buffer.from(options.diffPatchSeparated, 'base64')
+    .toString('utf-8')
+    .split('\n')
+    .slice(0, -1);
+  let highestChainID = false;
+  let additionalDeploymentToSameChainID = false;
+
+  for (const file of fileLineChangeJSON.files) {
+    const { path, additions, deletions } = file;
+    debug(`Checking ${path} with ${additions} additions and ${deletions} deletions`);
+    if (additions === 2 && deletions === 1) {
+      highestChainID = true;
+    } else if (additions === 1 && deletions === 1) {
+      additionalDeploymentToSameChainID = true;
+    } else if (additions !== 1 || deletions !== 0) {
+      throw new Error(`ERROR: ${path} has invalid changes`);
+    }
+  }
+
+  if (highestChainID) {
+    debug('Highest chain ID deployment');
+    let i = 0;
```

Comment:
Nit: now that we have a proper language, we can use a state enum here:

```
enum State {
  Removal,
  AddingComma,
  NewChain
}
```

Reply:
Updated 👍🏾